src/modules/dataGroup.js

```diff
@@ -3,69 +3,191 @@ import ClassNames from 'classnames';
+                                    <List.Item
+                                        key={`${bemClassName}_list_item-${index}`}
+                                    >
+                                        <DataGroupExpandedRow
+                                            bemClassName={bemClassName}
+                                            data={data}
+                                            row={row}
+                                        />
+                                    </List.Item>
+                                ))}
+                            </List>
+                        </div>
                     ))}
-                </List>
+                </div>
+            );
+        }
+
+        return (
+            <div
+                className={containerClasses}
+                id={id}
+                onClick={this.onToggleExpand}
+                role="presentation"
+                style={style}
+            >
+                {isExpandable && isExpanded && !_.isEmpty(expandableSections) && (
+                    <div className={`${bemClassName}_inner_container_loader`}>
+                        <ActivityIndicator
+                            color="backgroundColorStatic"
+                            size={32}
+                        />
+                    </div>
+                )}
+                <div
+                    className={`${bemClassName}_inner_container`}
+                >
+                    <div
+                        className={`${bemClassName}_header_section`}
+                        style={{
+                            marginBottom: '33px',
+                        }}
+                    >
+                        {header && (
+                            <Header
+                                className={`${bemClassName}_header_title`}
+                                weight="bold"
+                                size="large"
+                                style={{
+                                    margin: '0',
+                                    lineHeight: '16px',
+                                }}
+                            >
+                                {header}
+                            </Header>
+                        )}
+
+                        {isExpandable && !_.isEmpty(expandableSections) && (
+                            <div
+                                className={`${bemClassName}_header_icon_section`}
+                            >
+                                <Icon
+                                    className={`${bemClassName}_header_icon`}
+                                    type={isExpanded ? 'contract' : 'expand'}
+                                />
+                            </div>
+                        )}
+                    </div>
+                    {isExpandable && isExpanded && !_.isEmpty(expandableSections) ?
+                        expandableJSX : rowJSX}
```

Comment:
I'm thinking it would most likely be better to return both together instead of toggling back and forth. The reasoning behind this thought is because of being able to do some of the animations.

Reply:
Fixed